hazelcast/src/main/java/com/hazelcast/client/config/ClientFailoverConfig.java

```diff
@@ -79,27 +93,62 @@ public static ClientFailoverConfig load() {
         }
     }
 
+    /**
+     * Adds the client config to the end of the alternative client configurations list.
+     *
+     * @param clientConfig the ClientConfig to add
+     * @return this for chaining
+     */
     public ClientFailoverConfig addClientConfig(ClientConfig clientConfig) {
         validateClientConfig(clientConfig);
         clientConfigs.add(clientConfig);
         return this;
     }
 
+    /**
+     * Sets the count of connection retries by the client to the alternative clusters.
+     * <p>
+     * When this value is reached and the client still could not connect to a cluster,
+     * the client shuts down. Note that this value applies to the alternative clusters
+     * whose configurations are provided with the client element.
```

Comment:
@puzpuzpuz First of all, thanks for the pr. Not sure how did I miss Javadoc on a public API. 

It did not understand what you mean by `client element`. 
And I think we explain much more here. The term `connection retries` can easily confuse users. Similarly I would replace the same term in the javadoc of `getTryCount`. 

A suggestion from me:

Sets the count of attempts to connect to a cluster. For each alternative cluster, a client will try to connect to the cluster respecting related `ConnectionRetryConfig`. When the client can not connect a cluster, it will try to connect `tryCount` times going over the alternative client configs in a round-robin session. 
This is triggered at the start and also when the client disconnects from the cluster and can not connect back to it by exhausting attempts described in `ConnectionRetryConfig`. In that case, the client will continue from where it is left off in ClientConfig lists, and try the next one again in round-robin `tryCount` times.

Reply:
> It did not understand what you mean by `client element`.

That's a 1-to-1 copy from our ref manual. See: https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#configuring-without-cname

I'm going to apply proposed changes, but the same should be done with the ref manual, I guess. cc @Serdaro

Reply:
Addressed in 98cb759c1275a7e8c5a9f12a32b11bda12cb38cf

Reply:
I see. the meaning of the client element makes sense since it is talking about the `client` tag in the XML. 

 After reading the reference manual, I have realized that `tryCount` means trying the whole list `tryCount` times. 
After that, I read my suggestion, and not sure if I could make it clear also.